parser/parser/codegen/generate_diagnostics.cc

```diff
@@ -68,6 +68,9 @@ tuple<string, string> MESSAGES[] = {
     {"MissingCommaBetweenKwargs", "missing \\\",\\\" between keyword args"},
     {"CurlyBracesAroundBlockPass", "block pass should not be enclosed in curly braces"},
 
+    // Error recovery hints
+    {"DedentedEnd", "Hint: closing \\\"end\\\" token was not indented as far as {} token"},
```

Comment:
This phrasing strikes me as wrong: it's not that it's indented the wrong amount, it's that we associated it with a different construct than the user intended.

Compare [the similar situation in Rust](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=bda2211abfc8408a5f10d59a763cb592) or[ the error from Clang for the C/C++ case](https://gcc.godbolt.org/z/31aMe434Y), which I think do a better job of communicating what went wrong.

Reply:
Do you think we can iterate on this message? I generally agree with you but this PR is huge.

Reply:
I am down for iterating on this message, maybe we will even get complaints telling us what people would have preferred instead. :grin:

Reply:
Thanks for the links—i went with something that looks more similar to the rust error messages.